cache/carbonlink.go

```diff
@@ -133,13 +133,15 @@ func (listener *CarbonlinkListener) handleConnection(conn net.Conn) {
 
 		reqData, err := ReadCarbonlinkRequest(reader)
 		if err != nil {
+			conn.(*net.TCPConn).SetLinger(0)
```

Comment:
The break after the message jumps out of the for loop which causes the rest of the function to run and then return at which point the deferred Close runs. If I've got the flow wrong here please advise.